explorer/client/src/components/search/Search.tsx

```diff
@@ -1,85 +1,183 @@
+                    result?.length && prevIndex < result.length - 1
+                        ? prevIndex + 1
+                        : 0
+                );
+            }
+
+            // Press Up Key
+            if (event.keyCode === 38) {
+                event.preventDefault();
+                setResultIndex((prevIndex) =>
+                    prevIndex > 0
+                        ? prevIndex - 1
+                        : result?.length && result.length > 1
+                        ? result.length - 1
+                        : prevIndex
+                );
+            }
+        },
+        [result]
+    );
+
+    // Clicking Outside the Search Bar and Results should clear the search
+
+    useEffect(() => {
+        setResultIndex(0);
+        document.addEventListener('click', handleClickOutside, false);
+        document.addEventListener('keydown', handleKeyPress, false);
 
-    const [pleaseWaitMode, setPleaseWaitMode] = useState(false);
+        return () => {
+            document.removeEventListener('click', handleClickOutside, false);
+            document.removeEventListener('keydown', handleKeyPress, false);
+        };
+    }, [handleKeyPress]);
 
     const handleSubmit = useCallback(
         (e: React.FormEvent<HTMLFormElement>) => {
             e.preventDefault();
-            // Prevent empty search
-            if (!input.length) return;
-            setPleaseWaitMode(true);
 
-            // remove empty char from input
-            let query = input.trim();
+            if (result?.length && result.length >= 1) {
+                navigate(
+                    `../${result[resultIndex].category}/${encodeURIComponent(
+                        result[resultIndex].input
+                    )}`,
+                    {
+                        state: result[resultIndex].result,
+                    }
+                );
 
-            navigateWithUnknown(query, navigate, network).then(() => {
+                setResult(null);
                 setInput('');
-                setPleaseWaitMode(false);
-            });
+            }
+        },
+        [navigate, result, resultIndex]
+    );
+    const handleOptionClick = useCallback(
+        (entry: ResultType) => () => {
+            navigate(
+                `../${entry.category}/${encodeURIComponent(entry.input)}`,
+                {
+                    state: entry.result,
+                }
+            );
+            setResult(null);
+            setInput('');
         },
-        [input, navigate, setInput, network]
+        [navigate]
     );
 
     const handleTextChange = useCallback(
-        (e: React.ChangeEvent<HTMLInputElement>) =>
-            setInput(e.currentTarget.value),
-        [setInput]
+        (e: React.ChangeEvent<HTMLInputElement>) => {
+            setInput(e.currentTarget.value);
+            if (!e.currentTarget.value) {
+                setResult(null);
+            } else {
+                Promise.all(
```

Comment:
This would be a better fit for a `useEffect` that depends on `input` changing.

Reply:
Implemented